x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/FunctionInfo.java

```diff
@@ -29,6 +29,11 @@
      */
     String description() default "";
 
+    /**
+     * Detailed descriptions of the function rendered in the docs.
+     */
+    String detailedDescription() default "";
```

Comment:
Added a new field for functioning annotation, as some functions have more complicated descriptions with several paragraphs or tables in it, e.g. bucket, date_diff.